front/lib/api/oauth.ts

```diff
@@ -373,6 +386,34 @@ export async function createConnectionAndGetSetupUrl(
     });
   }
 
+  // For Salesforce, move client_id and client_secret to related_credential
+  let relatedCredential:
+    | {
+        content: Record<string, unknown>;
+        metadata: { workspace_id: string; user_id: string };
+      }
+    | undefined = undefined;
+
+  const clientId: string | undefined = extraConfig.client_id;
+
+  if (provider === "salesforce") {
```

Comment:
This should be a PROVIDER_STRAGEY thing. No provider conditional in main oauth flow 🙏

Reply:
That makes sense 👍

Reply:
Done